imaplugin/src/main/java/com/kaltura/playkit/plugins/ima/IMAPlugin.java

```diff
@@ -1079,7 +1080,21 @@ public void run() {
                 break;
             case CLICKED:
                 isAdIsPaused = true;
-                messageBus.post(new AdEvent(AdEvent.Type.CLICKED));
+                String clickThruUrl = null;
```

Comment:
Consider extracting this logic to a new method `handleAdClicked()`. IMHO it's too long for a `case`.